src/ProjectNeon/Assets/Features/Effects/CostPrimaryResourceEffect.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using UnityEngine;
+
+public sealed class CostPrimaryResourceEffect : Effect
+{
+    private Effect _effect;
+    private int _cost;
+
+    public CostPrimaryResourceEffect(Effect effect, int cost)
+    {
+        _effect = effect;
+        _cost = cost;
+    }
+
+    void Effect.Apply(Member source, Target target)
+    {
+        _effect.Apply(source, target);
+        source.State.SpendPrimaryResource(_cost);
```

Comment:
How do we ensure that we don't perform this action if the cost can't be paid?

Reply:
@SilasReinagel I think that it shouldn't be made here. The UI should't even let us pick a card if we didn't have the enough resources to play it. If we instantiated this cost effect then we already have the sufficient resource to play it

Reply:
@paulodamaso This is the cost of a triggered effect, not the cost of the card. The card has already been played and paid for.

Nothing can predict how many time Counter-Attack will trigger. The Solider is allowed to Counter each attack, until he runs out of ammo. Then, obviously, he isn't allowed to keep counter-attacking with his empty gun...